framework/PageToolbar/PageToolbarFilter.tsx

```diff
@@ -266,10 +281,49 @@ function ToolbarFilterComponent(props: {
     [filter.key, setFilterState]
   );
 
+  let isHasOrFilter = false;
+  if (filterState) {
+    for (const key in filterState) {
+      if (key === filter.key) continue;
+      const filterValues = filterState[key];
+      if (filterValues && filterValues?.length >= 2) {
+        isHasOrFilter = true;
+        break;
+      }
+    }
+  }
+
   switch (filter.type) {
-    case ToolbarFilterType.Text:
+    case ToolbarFilterType.SingleText:
       return (
-        <ToolbarTextFilter
+        <ToolbarSingleTextFilter
+          key={filter.key}
+          id={props.id ?? filter.key}
+          placeholder={filter.placeholder}
+          comparison={filter.comparison}
+          setValue={(value) => setFilterValues(() => (value ? [value] : []))}
+          value={filterValues && filterValues?.length > 0 ? filterValues[0] : ''}
+          hasKey={!!filterState?.[filter.key]}
+        />
+      );
+
+    case ToolbarFilterType.MultiText:
+      if (isHasOrFilter && props.limitFiltersToOneOrOperation) {
```

Comment:
When I turn limitFiltersToOneOrOperation to true, isHasOrFilter is false every time, so I still have multi version. Is it because in PR description, there is note about that it will be used for AWX?

Because isHasOrFilter is evaluated to false when I have active filter, thus the condition is skipped and  limitFiltersToOneOrOperation  does not work.

Reply:
This is working for me. Updated documentation for
limitFiltersToOneOrOperation

```
  /**
   * Limits the filters so that only one filter can be set to an OR operation.
   *
   * Example: AWX can either have an OR on type or status but not both.
   * So once one has 2 selections, the other becomes a single select.
   * Example: (Status is pending or success) and type is inventory.
   */
```

Reply:
Add tests that verify this functionality.

Reply:
If I understand it correctly, it should automatically convert all multiselects to single select when this is set?

Try pls that in collections.tsx:
      <PageTable<CollectionVersionSearch>
        id="hub-collection-version-search-table"
        toolbarFilters={toolbarFilters}
        toolbarActions={toolbarActions}
        tableColumns={tableColumns}
        rowActions={rowActions}
        errorStateTitle={t('Error loading collections')}
        emptyStateTitle={t('No collections yet')}
        emptyStateDescription={t('To get started, upload a collection.')}
        emptyStateButtonText={t('Upload collection')}
        emptyStateButtonClick={() => pageNavigate(HubRoute.UploadCollection)}
        {...view}
        defaultTableView="list"
        defaultSubtitle={t('Collection')}
        limitFiltersToOneOrOperation={true}
      />
      
      But in the page, I still can select multiple filter:
      
![image](https://github.com/ansible/ansible-ui/assets/23277791/ed7f5ec9-b2a1-4f4b-9355-522cb2c56e21)

Reply:
Oh, ok, one or operation, I though it should disable any or operation :-) Thus convert multiselect to single select. That makes sense. Approving.